service/authorization/v2/cache.go

```diff
@@ -0,0 +1,300 @@
+package authorization
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/opentdf/platform/protocol/go/policy"
+	otdfSDK "github.com/opentdf/platform/sdk"
+	"github.com/opentdf/platform/service/internal/access/v2"
+	"github.com/opentdf/platform/service/logger"
+	"github.com/opentdf/platform/service/pkg/cache"
+)
+
+const (
+	attributesCacheKey          = "attributes_cache_key"
+	subjectMappingsCacheKey     = "subject_mappings_cache_key"
+	registeredResourcesCacheKey = "registered_resources_cache_key"
+)
+
+var (
+	// Cache tags for authorization-related data set in the cache
+	authzCacheTags = []string{"authorization", "policy", "entitlements"}
+
+	// stopTimeout is the maximum time to wait for the periodic refresh goroutine to stop
+	stopTimeout = 5 * time.Second
+
+	// valid minimum refresh interval for the cache (too frequently may overload policy services)
+	minRefreshInterval = 15 * time.Second
+	maxRefreshInterval = 1 * time.Hour
```

Comment:
Why do we need to set a max? Shouldn't this be up to the policy admin if they want to only refresh the policy cache once a day.

Reply:
Its probably better to keep it more restrictive to start but I could see a higher max with how infrequent we have seen policy actually changing.

Reply:
Agreed. We can definitely continue to monitor and adjust over time.